server/controller/routes/admin/viewEventsOnDate.js

```diff
@@ -3,19 +3,20 @@ const moment = require('moment');
 const { getEventsDate } = require('../../../database/queries/events');
 
 const today = moment().format('YYYY-MM-DD');
-const viewEventsOnDate = (req, res, next) => {
-  getEventsDate()
-    .then(({ rows }) => {
-      if (rows.length !== 0) {
-        const events = rows.filter(
-          (el) => moment(el.date).format('YYYY-MM-DD') === today
-        );
-        res.json(events);
+
+const viewEventsOnDate = async (req, res, next) => {
+  try{
+    const {rows}=await getEventsDate()
+      if (!rows.length) {
```

Comment:
fix the condition here : 
```suggestion
      if (rows.length ) {
```